packages/shared/src/hooks/useFeedSettings.ts

```diff
@@ -57,6 +102,38 @@ export default function useFeedSettings(): FeedSettingsReturnType {
     }, 100);
   }, [tagsCategories, feedSettings, avoidRefresh]);
 
+  useEffect(() => {
+    const isFeedQueryEmpty = isObjectEmpty(feedQuery);
+    const isFeedSettingsEmpty = isObjectEmpty(feedSettings);
+    if (
+      !isFeedSettingsEmpty ||
+      !loadedUserFromCache ||
+      isFeedQueryEmpty ||
+      !shouldShowMyFeed
+    ) {
+      return;
+    }
+
+    /*
+      This sets the initial value for the `feedSettings` property
+      The query that fetches the information related to this property doesn't include this mentioned prop if not logged in - results to be `undefined`
+      This also covers to initially load the filters left by the user if they have not finished the registration and came back
+    */
+
+    const localFeedSettings = getLocalFeedSettings();
+    client.setQueryData<AllTagCategoriesData>(filtersKey, (current) => ({
+      ...current,
+      feedSettings: { ...localFeedSettings },
+    }));
+  }, [feedQuery, feedSettings, loadedUserFromCache, user]);
+
+  useEffect(() => {
+    if (!hasClearedCache && loadedUserFromCache && user) {
+      hasClearedCache = true;
+      storage.removeItem(LOCAL_FEED_SETTINGS_KEY);
+    }
+  }, [loadedUserFromCache, user]);
```

Comment:
Reading your comment I'm not sure we need this snippet.
The state we keep locally is used just as a way to sync the settings back once the user is registered.
If they never finish registration we should not show the preselected values in there every time they load the app.

Reply:
Totally makes sense, let me fix this up quickly!